Mindstack-Technologies/inprime-admin
# Enforce ESLint for Code Quality: Linting and Pre-Commit Hooks

## Description

This pull request introduces ESLint to the project, enforcing a consistent code style and preventing the inclusion of dead or unused code during commits. The changes include the following:

1. **ESLint Configuration:** Added ESLint configuration files to define linting rules for maintaining code quality.

2. **Husky Hooks:** Integrated Husky to run ESLint checks before each commit, ensuring that committed code adheres to the defined coding standards.

3. **npm Script:** Added an npm script to run ESLint checks manually, providing flexibility for developers to lint their code on demand.

## Changes Made

- Added ESLint configuration files (.eslintrc.js) to define linting rules.
- Integrated Husky to run ESLint on pre-commit hooks.
- Added an npm script for manual ESLint checks.

## How to Test

1. Clone this branch locally.
2. Make changes to the code intentionally introducing linting errors.
3. Attempt to commit the changes and observe ESLint preventing the commit.

## Why

By enforcing ESLint, we aim to enhance code quality, maintainability, and collaboration across the development team. Linting helps catch potential issues early in the development process, leading to a more robust and consistent codebase.

Please review and provide feedback. Thank you!
